### ISSUE TITLE
General Statistics File Count and LoC do not match actual values

### ISSUE THREAD

**Issue by GoogleCodeExporter** [waterxi/svnplot#63]
```
What steps will reproduce the problem?
1. Run svnlog2sql over entire repository
2. run svnplot over /trunk.  Under General Statistics shows File Count = 349, 
LoC = 78280
3. export head of /trunk from svn
4. count files using 'find trunk -type -f | wc -l' shows 419 files
5. count lines using 'find trunk -type -f | xargs wc -l' shows 91433 lines

What is the expected output? What do you see instead?
I expected the File Count & LoC to match.

What version of the product are you using? On what operating system?
7.3 on Windows XP

Please provide any additional information below.
I also noticed that the File Type graph does not show all the files.  In my 
case there are 16 *.jsp files on the /trunk, however the File Types chart does 
show now any.  Also, there are 146 *.xml files on the /trunk but the File Types 
chart only shows 105.

It just appears that files are being skipped.
```

Original issue reported on code.google.com by `refor...@gmail.com` on 14 Jan 2011 at 7:11

**Comment by GoogleCodeExporter** [comment]
```
The issue may be related to how SVN records additions vs deletions.  The 
following is a snippet from a real svn log.  At r111 a directory full of files 
was added.  All files are recorded separately in the log.  At revision r113 the 
entire directory was deleted.  Note only the directory is in the log, not the 
deletion of the files it contains.  The problem is compounded by revision r118 
where the directory of files is added again.  These files are getting counted 
twice.

------------------------------------------------------------------------
r118 | tuhin.banerjee | 2008-09-23 07:01:52 -0400 (Tue, 23 Sep 2008) | 1 line
Changed paths:
   A /trunk/Config-Files/All_Environment
   A /trunk/Config-Files/All_Environment/Document
   A /trunk/Config-Files/All_Environment/Document/CVS IDM -SelfservicepasswordComponent Listing V0.2.doc
   A /trunk/Config-Files/All_Environment/Document/CVS-IDM -Test Scriptsversion3.doc
   A /trunk/Config-Files/All_Environment/Document/CVS-IDM Deployment phase (1+2).doc
   A /trunk/Config-Files/All_Environment/Forms
   A /trunk/Config-Files/All_Environment/Forms/CVS Change Password Form.xml
   A /trunk/Config-Files/All_Environment/Forms/CVS Change User Answers Form.xml
. . .
------------------------------------------------------------------------
r113 | tuhin.banerjee | 2008-09-23 06:06:19 -0400 (Tue, 23 Sep 2008) | 1 line
Changed paths:
   D /trunk/Config-Files/All_Environment
. . .
------------------------------------------------------------------------
r111 | tuhin.banerjee | 2008-09-22 12:40:04 -0400 (Mon, 22 Sep 2008) | 1 line
Changed paths:
   A /trunk/Config-Files/All_Environment
   A /trunk/Config-Files/All_Environment/Document
   A /trunk/Config-Files/All_Environment/Document/CVS IDM -SelfservicepasswordComponent Listing V0.2.doc
   A /trunk/Config-Files/All_Environment/Document/CVS-IDM -Test Scriptsversion3.doc
   A /trunk/Config-Files/All_Environment/Document/CVS-IDM Deployment phase (1+2).doc
   A /trunk/Config-Files/All_Environment/Forms
   A /trunk/Config-Files/All_Environment/Forms/CVS Change Password Form.xml
   A /trunk/Config-Files/All_Environment/Forms/CVS Change User Answers Form.xml




```

Original comment by `refor...@gmail.com` on 14 Jan 2011 at 9:01

**Comment by GoogleCodeExporter** [comment]
```
Your observation about how svn records addition/deletions is correct. SVNplot 
tries to get around this problem by 'creating dummy entries' for the files when 
the directory is deleted. 

Looks like There are still a few issues in dummy entry creation. I have a look 
again.
```

Original comment by `nitinbh...@gmail.com` on 15 Jan 2011 at 1:34

**Comment by GoogleCodeExporter** [comment]
Original comment by `nitinbh...@gmail.com` on 15 Jan 2011 at 1:34

**Comment by GoogleCodeExporter** [comment]
Original comment by `nitinbh...@gmail.com` on 28 May 2011 at 1:50
- Changed state: **Accepted**

**Comment by GoogleCodeExporter** [comment]
```
many line count stats related issues are fixed in 0.7.5. Please retest after 
upgrading to 0.7.5
```

Original comment by `nitinbh...@gmail.com` on 28 May 2011 at 1:50
- Changed state: **Fixed**
